### PR TITLE
defined matchers for rspec

### PR DESCRIPTION
**Pull request by alanhala**
### Summary

- Defined `be_paginated` and `be_paginated_with` matchers for rspec.
- Added specs that checks that both matchers works properly. I don't know if we need to add any other kind of specs for this new feature.

### CODE REVIEW

File: lib/wor/paginate/matchers/rspec.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+RSpec::Matchers.define :be_paginated_with do |formatter|
+  define_matcher_for self, formatter
+end
+
+RSpec::Matchers.define :be_paginated do
+  define_matcher_for self, Wor::Paginate::Formatter
+end
+
+def define_matcher_for(matcher, formatter)
+  formatter_matcher = Wor::Paginate::Matchers::FormatterMatcher.new(formatter)
```
Comment (alanhala) [inline lib/wor/paginate/matchers/rspec.rb]: I don't know we I cant just call FormatterMatcher just like I did with the MockedAdapter inside the FormatterMatcher class. Just tell me if you know a way to do that

File: spec/matchers_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    end
+
+    context 'receiving a custom formatter as an argument' do
+      it 'checks the json keys matches between the custom formatter and the response' do
+        expect(response_body(response)).to be_paginated_with(CustomFormatter)
+      end
+    end
+  end
+
+  describe '#be_paginated_with' do
```
Comment (icoluccio) [inline spec/matchers_spec.rb]: Isn't the context for this describe the same as the previous one? I'd extract all that to a common context and then use different befores.

File: lib/wor/paginate/matchers/formatter_matcher.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        def initialize(formatter)
+          @formatted_keys = formatter.new(MockedAdapter.new).format.as_json.keys
+        end
+
+        def define_matcher_for(matcher)
+          # Needed to save this variable because we lose the object scope inside the block
+          formatted_keys = @formatted_keys
+          matcher.match { |actual_response| actual_response.keys == formatted_keys }
+          matcher.failure_message do |actual_response|
+            "expected that #{actual_response.keys} would be equal to #{formatted_keys}"
```
Comment (icoluccio) [inline lib/wor/paginate/matchers/formatter_matcher.rb]: Shouldn't the message be less general? Something like "expect X to be paginated with keys Y"

File: lib/wor/paginate/matchers/rspec.rb (hunk 1 above)
Comment (icoluccio) [inline lib/wor/paginate/matchers/rspec.rb]: I think it's clear this way anyway

File: spec/matchers_spec.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      end
+    end
+
+    before do
+      get :index_custom_formatter
+    end
+
+    context 'receiving a custom formatter as an argument' do
+      it 'checks the json keys matches between the custom formatter and the response' do
+        expect(response_body(response)).to be_paginated_with(CustomFormatter)
```
Comment (icoluccio) [inline spec/matchers_spec.rb]: We should test failure as well

**Pull request closed without merging**